### ISSUE TITLE
Basic Node and Express hard to follow

### ISSUE THREAD

**Issue by zeuslawyer** [freeCodeCamp/freeCodeCamp#17754]
#### Describe your problem and - if possible - how to reproduce it
As has been noted by a few users, FCC's remarkably clear and systematic approach breaks a bit jarringly when we start the Express Section.  NPM / Package management was fantastic. But the simple express server stuff presumes a lot of knowledge/experience that is not beginner friendly.

see similar comments: https://forum.freecodecamp.org/t/whats-with-the-basic-node-and-express-intro/196525/4

#### Add a Link to the page with the problem
https://learn.freecodecamp.org/apis-and-microservices/basic-node-and-express/serve-an-html-file

in the above, for example, it goes straight into some global variable called __dirname without any explanation or context.  

#### Tell us about your browser and operating system

* Browser Name: 
* Browser Version: 
* Operating System:

**Comment by thedtj** [comment]
Another area for confusion in this section is the [Get Query Parameter Input from the Client](https://learn.freecodecamp.org/apis-and-microservices/basic-node-and-express/get-query-parameter-input-from-the-client) challenge. A POST method is required to complete the challenge, but POST methods aren't explained until the next challenge [Use body-parser to Parse POST Requests](https://learn.freecodecamp.org/apis-and-microservices/basic-node-and-express/use-body-parser-to-parse-post-requests).

I can open a new issue for if this would be more appropriate.

**Comment by kiwisquash** [comment]
RE: Get Query Parameter Input from the Client challenge

I was able to do the Get Query Parameter Input from the Client challenge without using the POST method. I came here to open an issue saying that the TIP was confusing, since POST is not needed for this challenge...

**Comment by RandellDawson** [comment]
As there has been no further discussion, we are closing this issue as stale.

If anyone is interested in continuing this discussion or has ideas of how best to make the desired improvements, you can request this issue to be reopened. Thanks and Happy Coding!
